app/custom_auth.py

```diff
@@ -0,0 +1,74 @@
+from django.contrib import auth
+from django.conf import settings
+from django.contrib.auth.models import User
+
+class CustomSSOMiddleware:
```

Comment:
Needs tests so that the code coverage doesn't drop below 90% ([see build failure](https://travis-ci.org/CatalystCode/doccano/builds/555958628#L1458)). I'd also move this class to `server/middlewares.py` to more closely follow Django conventions.

Reply:
The build is passing now: https://travis-ci.org/CatalystCode/doccano/builds/556378487?utm_source=github_status&utm_medium=notification